### PR TITLE
Add support for `autoLayout` in email configuration profile

### PR DESCRIPTION
**Pull request by gerhard-lechner**
For details pls see issue #16229

### CODE REVIEW

File: src/Mailer/Mailer.php [hunk 1]
```diff
@@ -459,6 +459,12 @@ public function setProfile($config)
             $this->viewBuilder()->setVars($config['viewVars']);
             unset($config['viewVars']);
         }
+        if (array_key_exists('autoLayout', $config)) {
```
Comment (dereuromark) [inline src/Mailer/Mailer.php]: This should be using isset() as null is not a useful value

**Comment by markstory** [comment]
I've added a test to your branch.

**Comment by markstory** [comment]
Docs changes are in cakephp/docs#7287

File: src/Mailer/Mailer.php (hunk 1 above)
Comment (othercorey) [inline src/Mailer/Mailer.php]:
```suggestion
        if (isset($config['autoLayout']) {
```

File: src/Mailer/Mailer.php [hunk 2]
```diff
@@ -459,6 +459,12 @@ public function setProfile($config)
             $this->viewBuilder()->setVars($config['viewVars']);
             unset($config['viewVars']);
         }
+        if (isset($config['autoLayout']) {
```
Comment (othercorey) [inline src/Mailer/Mailer.php]:
```suggestion
        if (isset($config['autoLayout'])) {
```

**Comment by othercorey** [comment]
Added migration notes for Mailer changes.

**Pull request merged**